libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewSubscriber.java

```diff
@@ -54,4 +64,12 @@ public void onChanged(@Nullable Boolean shouldRecordScreenshot) {
       }
     });
   }
+
+  @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+  void unsubscribe() {
+    navigationViewModel.retrieveRoute().removeObservers(lifecycleOwner);
```

Comment:
i dont think this is necessary considering the lifecycle is tied to this view anyway